src/node_worker.cc

```diff
@@ -560,14 +514,12 @@ void Worker::Unref(const FunctionCallbackInfo<Value>& args) {
 
 void Worker::Exit(int code) {
   Mutex::ScopedLock lock(mutex_);
-
   Debug(this, "Worker %llu called Exit(%d)", thread_id_, code);
-  if (!thread_stopper_.IsStopped()) {
+  if (env_ != nullptr) {
     exit_code_ = code;
-    Debug(this, "Received StopEventLoop request");
-    thread_stopper_.Stop();
-    if (isolate_ != nullptr)
-      isolate_->TerminateExecution();
+    env_->ExitEnv();
```

Comment:
Maybe we could use the public `Stop(env_);` here?

I’d love it if we could use more public APIs here in general, and avoid internal ones altogether. That helps us get a better overview over what “wires” we still have to expose to allow embedders to do something like this.

Reply:
done, routed through `node::Stop`. One caveat here, that now we cannot do `CHECK(env->is_main_thread())` as there are workers which call this in their `process.exit` sequence. I guess that is fine, as per your next comment?